thewalrus/_permanent.py

```diff
@@ -15,29 +15,24 @@
 Permanent Python interface
 """
 import numpy as np
-
+from numba import jit
 from ._hafnian import hafnian_repeated
-from .libwalrus import perm_complex, perm_real, perm_BBFG_real, perm_BBFG_complex
 
+# pylint: disable = C0103, R0914
 
-def perm(A, quad=True, fsum=False, method="bbfg"):
-    """Returns the permanent of a matrix using the `method` formula
 
+def perm(A, method="bbfg"):
+    """Returns the permanent of a matrix using the `method` formula
     For more direct control, you may wish to call :func:`perm_real`
     or :func:`perm_complex` directly.
-
     Args:
         A (array): a square array.
-        quad (bool): If ``True``, the input matrix is cast to a ``long double``
-            matrix internally for a quadruple precision hafnian computation.
```

Comment:
out of curiosity, is the numba implementation only double precision? Is there any disadvantage to losing quad precision and fsum?

Reply:
It is double precision. I don't think you gained much from `fsum`. As for the `quad` you might have gained a little bit more precision, but not too much, since `quad` simply means 80 instead of 64.